exla/Makefile

```
@@ -40,7 +40,28 @@ endif
 
 LDFLAGS = -L$(XLA_EXTENSION_LIB) -lxla_extension -shared
 
-ifeq ($(shell uname -s), Darwin)
+ifeq ($(CROSSCOMPILE),)
+	# Interrogate the system for local compilation
+	UNAME_S = $(shell uname -s)
+
+	NVCC_RESULT := $(shell which nvcc 2> /dev/null)
+	NVCC_TEST := $(notdir $(NVCC_RESULT))
+
+ifeq ($(NVCC_TEST),nvcc)
+	NVCC := nvcc
+	NVCCFLAGS += -DCUDA_ENABLED
+else
+	NVCC := $(CXX)
+	NVCCFLAGS = $(CFLAGS)
+endif
+else
+	# Determine settings for cross-compiled builds like for Nerves
+	UNAME_S = Linux
+	NVCC := $(CXX)
+	NVCCFLAGS = $(CFLAGS)
+endif
```

I believe `UNAME_S` should be given as an optional env flag. Linux is _probably_ always the target, but I don't think we can assert this for sure. Maybe a different name should be used for the env var as it's now gonna be (optionally) exposed.

I also wonder if we should allow NVCC to be provided via the toolchain for systems like the nvidia jetson. 

```suggestion
else
	# Determine settings for cross-compiled builds like for Nerves
	UNAME_S ?= Linux
	NVCC := $(CXX)
	NVCCFLAGS = $(CFLAGS)
endif
```

---

Actually, given that this is only used for checking if it's mac or not, we can merge as is. Crosscompiling for mac is highly unlikely IMO

---

The only cross-compilation I've run into that hasn't been Linux is for RTOSs and the "Linux" options seem to work. It might be easier to address when it comes up. 

Thanks for merging!